common/src/main/java/io/pravega/common/concurrent/Futures.java

```diff
@@ -44,6 +45,31 @@
  */
 public final class Futures {
 
+    /**
+     * Makes sure that the result future given out to caller is actually completed on caller's executor.
+     * So any chaining, if done without specifying an executor, will either happen on
+     * caller's executor or fork join pool but never on someone else's executor.
+     *
+     * @param future The future to execute.
```

Comment:
The future is not "to execute". The future is some object that will be completed after execution. Chances are that the execution of the backing task is already underway.

Reply:
Fixed.